src/main/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.darts.cases.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.cases.mapper.GetCasesMapper;
+import uk.gov.hmcts.darts.cases.model.GetCasesRequest;
+import uk.gov.hmcts.darts.cases.model.ScheduledCase;
+import uk.gov.hmcts.darts.cases.service.CaseService;
+import uk.gov.hmcts.darts.common.api.CommonApi;
+import uk.gov.hmcts.darts.common.entity.Hearing;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CaseServiceImpl implements CaseService {
+
+    private final HearingRepository hearingRepository;
+    private final CommonApi commonApi;
+
+    @Override
+    public List<ScheduledCase> getCases(GetCasesRequest request) {
+
+        List<Hearing> hearings = hearingRepository.findByCourthouseCourtroomAndDate(
+            request.getCourthouse(),
+            request.getCourtroom(),
+            request.getDate()
+        );
+        createCourtroomIfMissing(hearings, request);
```

Comment:
This right, creating new courtrooms of the back of a GET request?

Reply:
yeah, i checked with Dave